tests/python/pants_test/backend/python/tasks/test_ctypes_integration.py

```diff
@@ -0,0 +1,71 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import glob
+import os
+from zipfile import ZipFile
+
+from pants.backend.native.config.environment import Platform
+from pants.option.scope import GLOBAL_SCOPE_CONFIG_SECTION
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import is_executable
+from pants.util.process_handler import subprocess
+from pants_test.backend.python.tasks.python_task_test_base import name_and_platform
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+def invoke_pex_for_output(pex_file_to_run):
+  return subprocess.check_output([pex_file_to_run], stderr=subprocess.STDOUT)
+
+
+class CTypesIntegrationTest(PantsRunIntegrationTest):
+
+  _binary_target = 'testprojects/src/python/python_distribution/ctypes:bin'
+
+  def test_run(self):
+    pants_run = self.run_pants(command=['run', self._binary_target])
+    self.assert_success(pants_run)
+
+    # This is the entire output from main.py.
+    self.assertIn('x=3, f(x)=17', pants_run.stdout_data)
+
+  def test_binary(self):
+    with temporary_dir() as tmp_dir:
+      pants_run = self.run_pants(command=['binary', self._binary_target], config={
+        GLOBAL_SCOPE_CONFIG_SECTION: {
+          'pants_distdir': tmp_dir,
+        }
+      })
+
+      self.assert_success(pants_run)
+
+      # Check for the pex and for the wheel produced for our python_dist().
+      pex = os.path.join(tmp_dir, 'bin.pex')
```

Comment:
Would be good to wrap this pants run in a try/except and clean up the output pex after the test finished (successfully or otherwise).

Reply:
I think that's handled by the `temporary_dir()` contextmanager? Because we use that for the `dist/` directory. Is something staying behind when you run the test?